app/flatpak-builtins-uninstall.c

```diff
@@ -153,7 +153,7 @@ flatpak_builtin_uninstall (int argc, char **argv, GCancellable *cancellable, GEr
   g_option_context_set_translation_domain (context, GETTEXT_PACKAGE);
 
   if (!flatpak_option_context_parse (context, options, &argc, &argv,
-                                     FLATPAK_BUILTIN_FLAG_STANDARD_DIRS,
+                                     FLATPAK_BUILTIN_FLAG_ALL_DIRS,
```

Comment:
Any reason we don't want OPTIONAL_REPO here? Is some other part of the implementation not robust to the "no-repo" case?

Reply:
it is not exactly clear to me what the requirements are for putting OPTIONAL_REPO there

Reply:
OPTIONAL_REPO means we don't fail with EPERM in the case of non-existing /var/lib/flatpak. We should properly handle that case for all operations that could realistically finish fine if the repo did exist and was empty. For example "list" or "get config", but not that depend on something being installed or a named remote being configured.

Reply:
If we add it we need to also ensure that the codepath handles the dir->repo == NULL case properly.

Reply:
it worked in my testing, for the fuzzy search part. I guess thats the only relevant part, since we won't find a match in an empty repo